File: 김찬우/프로그래머스 /Lv.4/도둑질.java
import java.lang.Math;

class Solution {
    public int solution(int[] money) {
        
       int mlen = money.length;
        
        int[] first = new int[mlen];
        int[] second = new int[mlen];
            
        for(int i = 0; i<mlen; i++){
            first[i] = money[i]; 
            second[i] = money[i]; 
        }
        
        // 3번째 까지는 미리 넣기 
        first[0] = money[0];
        first[1] = 0;
        first[2] +=first[0];
        second[0] = 0;
        second[1] = money[1];

        for(int j = 3; j<mlen;j++){
            
            first[j] += Math.max(first[j-2],first[j-3]);
            second[j] += Math.max(second[j-2],second[j-3]);
            
        }
        
        int firstMax = Math.max(first[mlen-2],first[mlen-3]);
        int secondMax = Math.max(second[mlen-1],second[mlen-2]);
        
        
       return Math.max(firstMax,secondMax);
        
        
    }
}

File: 이건희/level4/도둑질.py
def solution(money):
    dp1 = [0 for _ in range(len(money))]
    dp2 = dp1.copy()
    dp1[:2] = [money[0], max(money[:2])]
    dp2[:2] = [0, money[1]]
    
    for i in range(2, len(money) - 1):
        dp1[i] = max(dp1[i - 2] + money[i], dp1[i - 1])
        
    for i in range(2, len(money)):
        dp2[i] = max(dp2[i - 2] + money[i], dp2[i - 1])

    return max(max(dp1), max(dp2))
